qcodes/instrument/visa.py

```diff
@@ -231,3 +232,26 @@ def snapshot_base(self, update: bool=False,
         snap['timeout'] = self.timeout.get()
 
         return snap
+
+    def wait_till_complete(self, check_after = 5):
+        """
+        WARNING: This function repeatedly clears the Event Status Register (ESR).
+
+        The *OPC? command is used to query the machine when it completes an operation.
+        However, in some cases the operation is too long. There are 2 ways of dealing with this.
+
+        1. Change the timeout to an appropriate value (if you can determine how long the operation will take.)
+           This may cause issues while debugging.
```

Comment:
what issues may it cause exactly? I'm not an expert (yet) in visa instruments, but so far as I see, the inconvenience of this "way" is that you are changing the timeout value for all the commands, while what you want is just make it longer for a very particular command. With that said, I think the docstring can be improved to be more explicit.